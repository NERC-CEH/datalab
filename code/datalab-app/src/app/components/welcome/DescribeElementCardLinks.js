import React from 'react';
import PropTypes from 'prop-types';
import Button from 'material-ui/Button';
import { CardActions } from 'material-ui/Card';

const Link = ({ displayName, href }) => (
  <Button style={{ fontSize: 'larger' }} onClick={() => window.open(href)} color="primary">
    {`${displayName} >`}
  </Button>
);

const DescribeElementCardLinks = ({ links }) => (
  <CardActions>
    {links.map(({ displayName, href }, idx) => (<Link key={`link-${idx}`} displayName={displayName} href={href} />))}
  </CardActions>
);

DescribeElementCardLinks.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    displayName: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
  })),
};

export default DescribeElementCardLinks;
