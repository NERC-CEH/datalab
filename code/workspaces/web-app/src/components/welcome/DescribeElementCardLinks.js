import React from 'react';
import PropTypes from 'prop-types';
import Button from '@mui/material/Button';
import CardActions from '@mui/material/CardActions';

const Link = ({ displayName, href }) => (
  <Button style={{ fontSize: 'larger' }} onClick={() => window.open(href)} color="primary">
    {`${displayName} >`}
  </Button>
);

const DescribeElementCardLinks = ({ links }) => (
  <CardActions>
    {links.map(({ displayName, href }, idx) => (<Link key={`link-${idx}`} displayName={displayName} href={href} />))}
  </CardActions>
);

DescribeElementCardLinks.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    displayName: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
  })),
};

export default DescribeElementCardLinks;
