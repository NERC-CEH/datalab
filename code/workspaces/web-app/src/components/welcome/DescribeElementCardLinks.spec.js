import React from 'react';
import { shallow } from 'enzyme';
import DescribeElementCardLinks from './DescribeElementCardLinks';

describe('DescribeElementCardLinks', () => {
  it('renders correct snapshot', () => {
    const links = [
      { displayName: 'displayName', href: 'address' },
      { displayName: 'anotherDisplayName', href: 'anotherAddress' },
    ];

    expect(shallow(<DescribeElementCardLinks links={links} />)).toMatchSnapshot();
  });
});
