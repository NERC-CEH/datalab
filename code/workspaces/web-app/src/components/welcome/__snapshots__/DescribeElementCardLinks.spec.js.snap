// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescribeElementCardLinks renders correct snapshot 1`] = `
<div>
  <div
    class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-4dagr2-MuiButtonBase-root-MuiButton-root"
      style="font-size: larger;"
      tabindex="0"
      type="button"
    >
      displayName &gt;
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-4dagr2-MuiButtonBase-root-MuiButton-root"
      style="font-size: larger;"
      tabindex="0"
      type="button"
    >
      anotherDisplayName &gt;
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
